google/cloud/sql/connector/instance.py

```diff
@@ -119,18 +119,18 @@ def __init__(
                 f"({ssl.OPENSSL_VERSION}), falling back to TLSv1.2\n"
                 "Upgrade your OpenSSL version to 1.1.1 for TLSv1.3 support."
             )
-            self.context.minimum_version = ssl.TLSVersion.TLSv1_2
-        self.expiration = expiration
+            context.minimum_version = ssl.TLSVersion.TLSv1_2
 
         # tmpdir and its contents are automatically deleted after the CA cert
         # and ephemeral cert are loaded into the SSLcontext. The values
         # need to be written to files in order to be loaded by the SSLContext
         with TemporaryDirectory() as tmpdir:
             ca_filename, cert_filename, key_filename = write_to_file(
-                tmpdir, server_ca_cert, ephemeral_cert, private_key
+                tmpdir, self.server_ca_cert, self.client_cert, self.private_key
             )
-            self.context.load_cert_chain(cert_filename, keyfile=key_filename)
-            self.context.load_verify_locations(cafile=ca_filename)
+            context.load_cert_chain(cert_filename, keyfile=key_filename)
```

Comment:
Are there other ways to load this data? If each connection has to read from the disk, this will result in worse performance.

Reply:
Maybe we could cache the SSL context so that we only load it once?

Reply:
I added caching the context for creating it only once per `ConnectionInfo`